include/common/qrack_types.hpp

```diff
@@ -13,19 +13,22 @@
 #pragma once
 
 #include <memory>
+#include <random>
 
 #define bitLenInt uint8_t
 #define bitCapInt uint64_t
 #define bitsInByte 8
+#define qrack_rand_gen std::mt19937_64
+#define qrack_rand_gen_ptr std::shared_ptr<std::mt19937_64>
```

Comment:
std::shared_ptr<qrack_rand_gen>

Reply:
Preprocessor is single pass, I'm almost positive, at least on defines. Even if it isn't, that's still the potential issue. You can always pivot on these two lines alone.